edb/edgeql/compiler/viewgen.py

```diff
@@ -922,11 +922,21 @@ def _normalize_view_ptr_expr(
             specified_required = shape_el.required
```

Comment:
The logic flow here still seems a little weird; why is all the required stuff conditioned on the base cardinality?

Reply:
Because if we don't know the upper bound, then we don't know the lower either.